machine/v1alpha1/types_openstack.go

```diff
@@ -366,3 +372,62 @@ type RootVolume struct {
 	// Deprecated: deviceType will be silently ignored. There is no replacement.
 	DeprecatedDeviceType string `json:"deviceType,omitempty"`
 }
+
+// blockDeviceStorage is the storage type of a block device to create and
+// contains additional storage options.
+type BlockDeviceStorage struct {
```

Comment:
do we need union discriminator in here? since both fields are optional...

Reply:
This is the union, so this one needs a `+union` on the struct and the Type field needs to be marked up as the union discriminator